Improve README

SonQBChau opened the pull request:
## What's new
- Updated README with instructions on Playwright installation, environment variable setup, and running tests (UI and API).
- Clarified test folder structure and naming conventions for UI and API tests.
- Added test checklist with expected values.
- Provided steps for updating tests and accessing test reports.

[DIG-1805](https://candig.atlassian.net/browse/DIG-1805)

[DIG-1805]: https://candig.atlassian.net/browse/DIG-1805?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

SonQBChau committed:
Update README.md

mshadbolt commented on README.md:
````diff
@@ -1,30 +1,134 @@
+Clone the repository and execute the following commands:
+
 ```bash
 npm install
````
```suggestion
cd playwright-runbook
npm install
```

SonQBChau committed:
rename performance to api

SonQBChau committed:
Merge branch 'develop' into docs

SonQBChau committed:
add missing overview test

SonQBChau committed:
Update README.md

Co-authored-by: Marion <mshadbolt@users.noreply.github.com>

mshadbolt commented on README.md:
```diff
@@ -1,30 +1,137 @@
+
+### UI Tests
+
+The `ui-tests` folder contains tests that simulate user interactions with the data portal. These tests verify basic frontend functionality, such as:
```
```suggestion
The `ui-tests` folder contains tests that simulate user interactions with the data portal. They assume that the default test data from the integration tests is loaded into the running deployment before running, and complete tests based on being logged in as user2, who is authorized for the program `SYNTH_02`. These tests verify basic frontend functionality, such as:
```
Is the edit I made true?

SonQBChau replied:
I used the admin credentials for the api_tests. I don’t think it will affect the summary page. Courtney and I discussed this briefly, and I think it’s better addressed in a separate PR.

kcranston commented on README.md:
```diff
@@ -1,30 +1,137 @@
+# Prerequisites
```
A brief explanation of the purpose of this repository would be good at the top of the README. Also, a link to the notebook repo?

kcranston commented on README.md:
```diff
@@ -1,30 +1,137 @@
+For local testing, set the URL to `localhost` or `candig.docker.internal`. All the credentials can be found in the `CanDIGv2/tmp/keycloak` folder.
 
-Rename `.env.sample` to `.env` and modify the values
+# Running Tests
```
Some more detail about running the tests would be helpful. Does this assume you have a running local CanDIG stack? Do you need to run the playbook from a specific directory? Can you run it against a remote stack?

SonQBChau committed:
add purposes

SonQBChau committed:
add link to notebook repo

SonQBChau committed:
Update README.md

- add env instruction for local/dev/prod
- clarify playwright should be tested with candig server running
- clarify only Tyk tests are needed
- clarify the user of dataset and user2

SonQBChau committed:
switch local var to env var

SonQBChau commented:
I updated the readme to address the comments above

SonQBChau committed:
Update README.md

SonQBChau committed:
clean up

SonQBChau committed:
switch default to user2

SonQBChau committed:
update api test with base_url

SonQBChau committed:
update ui test using user2

SonQBChau committed:
update search filter for user2

SonQBChau committed:
clean up

SonQBChau commented:
I put some changes to clean up the PR and switch to user2 default

mshadbolt commented on README.md:
```diff
@@ -1,30 +1,156 @@
-# Running the Test
+## Running Tests
+
+Once the URL is set and the target server (local/dev/prod) is running, you can run all tests by using:
```
I think if this is not the way you expect people to run the tests, it should go after the ways below, rather than first

SonQBChau replied:
should I take it out?

mshadbolt reviewed (approved):
running tests works for me

SonQBChau committed:
remove run all test
